shell.nix

```diff
@@ -0,0 +1,76 @@
+{ pkgs ? import <nixpkgs> {} }:
+
+with pkgs;
+let
+  lib = import <nixpkgs/lib>;
+  inherit (lib) optional optionals;
+  postgresql = pkgs.postgresql_14;
+  make = pkgs.gnumake;
+  go = pkgs.callPackage ./nix/go.nix { pkgs=pkgs; };
+
+  basePackages = [
+    postgresql
+    go
+    pkgs.kubernetes-helm
+    pkgs.nodejs-14_x
+    pkgs.yarn
+    pkgs.jq
+    pkgs.minikube
+    make
+  ];
+
+  requiredPackages = basePackages
+    ++ lib.optional stdenv.isLinux inotify-tools
+    ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
+        CoreFoundation
+        CoreServices
+      ]);
+
+in pkgs.mkShell {
+    name = "dev";
+
+    buildInputs = requiredPackages;
+
+    shellHook = ''
+        # setup postgres
+        export PGDATA="$PWD/.yatai-db"
+        export PGHOST="$PWD"
+        export SOCKET_DIRECTORIES="$PWD/.sockets"
+        mkdir -p $PGDATA
+
+        if [[ ! $(grep listen_address $PGDATA/postgresql.conf) ]]; then
+            initdb -D $PGDATA
+            createuser postgres -h localhost
+            createdb yatai
+            cat >> "$PGDATA/postgresql.conf" <<-EOF
+listen_addresses = 'localhost'
+port = 5432
+unix_socket_directories = '$PGHOST'
+EOF
+        fi
+        pg_ctl -l $PGDATA/logfile start
+
+        function end {
+          echo "Shutting down the database..."
+          pg_ctl stop
+          echo "Removing directories..."
+          rm -rf $SOCKET_DIRECTORIES
+          minikube stop
+        }
+        trap end EXIT
+
+        # setup for dashboard
+        alias scripts='jq ".scripts" dashboard/package.json'
+
+        make fe-deps be-deps
+        minikube start --cpus 4 --memory 4096
+
+        # setup helm chart
```

Comment:
Yatai helm chart is for installing yatai in a production environment, if you are only developing yatai, I don't think you need to install yatai helm chart

Reply:
noted